app/src/main/java/com/github/orkest/ViewModel/profile/ProfileViewModel.kt

```diff
@@ -97,5 +97,102 @@ open class ProfileViewModel(private val user: String) : ViewModel() {
         }
     }
 
+    /**
+     * Function only called when visiting the profile of another user
+     * Checks whether the current logged in user follows this account
+     */
+    open fun isUserFollowed(): CompletableFuture<Boolean>{
+        if(user == Constants.currentLoggedUser){ Log.e(TAG, "Cannot call this function when visiting the current logged-in user's profile", IllegalArgumentException()) }
+        val future = CompletableFuture<Boolean>()
+        userDocument(user).get()
+            .addOnSuccessListener { document ->
+                if (document != null && document.exists()) {
+                    val user = document.toObject(User::class.java)
+                    if (user != null && user.followers.contains(Constants.currentLoggedUser)) {
+                        future.complete(true)
+                    } else{
+                        future.complete(false)
+                    }
+                }
+            }
+            .addOnFailureListener { e -> Log.w(TAG, "Error getting user data", e) }
+        return future

```

Comment:
Looks good to me, However I think I will need to refractor the database operations in a following commit, so that we can do all our database operation in the part of the code Dedicated to It and not in our viewmodels